### DISCUSSION TITLE
Resave an already saved local website with SingleFile

### DISCUSSION THREAD

**Discussion by feetee** [gildas-lormeau/SingleFile#1449] [Q&A]
Hello community,

how can I save a website that has already been saved locally again with SingleFile? For example, a completely saved website with HTML file and folder with images, CSS, JavaScript, fonts, etc. Ideally also web pages saved in MHT or MAFF format. The re-save should work as soon as Firefox has completely loaded the web page, right? Unfortunately, neither local images nor local CSS are recognized by SingleFile.

Greeting, feetee

**Comment by bennjmin** [comment]
I'm not a dev, but it sounds to me like SingleFile isn't meant to do that. I also assume you can still save any locally saved page as a single .mhtml file, straight from almost any browser.

**Reply by feetee** [reply]
Thank you for your prompt reply.
Too bad that this probably won't work. I would have liked to save my saved web pages in MAFF, MHT and normal format as individual HTML files, because SingleFile automatically removes superfluous CSS, comments etc. and saves disk space. Where can I tell the developer about this wish?

**Comment by gildas-lormeau** [comment]
You should be able to resave HTML and MAFF pages (by unzipping them first) in Firefox out of the box. In Chromium-based browsers, you need to enable `Allow access to file URLs` in the extension page chrome://extensions/?id=mpiodijhokgodhhofbcjdecpffjipkle. 
Alternatively, you can use the CLI tool https://github.com/gildas-lormeau/single-file-cli.

In order to resave MHTML pages, you would need to find a tool to transform them to pages with external resources in a sub-folder.

**Reply by feetee** [reply]
Thank you for your reply.

Of course I can resave HTML, MHTML and MAFF files directly to HTML via Firefox, but only with external resources in a subfolder, but I would like to do this with SingleFile because it optimizes the HTML and CSS sources and also only creates a single HTML file. Where can I find the corresponding option for "Allow access to file URLs" in Firefox, as about:config does not show it to me with these terms? I do not use Chrome and browsers based on it.

It shouldn't really matter to SingleFile whether the CSS, JavaScript, image sources etc. are accessible and retrievable via a URL or a file path or whether I want to annotate and save a web page already saved with SingleFile again with SingleFile and the web page no longer exists on the Internet, for example.

Does the Single-File-CLI tool do anything differently than the extension in Firefox? I don't want to install the Chrome browser just for this, especially if no other result is to be expected.

**Reply by gildas-lormeau** [reply]
Sorry for the confusion, I was meaning that you should be able with SingleFile in Firefox to save pages loaded from the filesystem with resources in a subfolder. If this does not work then you could use a simple HTTP server like [`http-server`](https://www.npmjs.com/package/http-server) for example. It would allow you to save pages from `http://localhost:8080`.

**Reply by feetee** [reply]
It is possible to (re)save web pages loaded from the file system with SingleFile, but without all subfolder resources. I will try your suggestion with the HTTP server.

**Reply by gildas-lormeau** [reply]
FYI, here's a list of alternatives to `http-server`: https://gist.github.com/willurd/5720255

**Reply by feetee** [reply]
Thank you.

**Comment by gildas-lormeau** [comment]
I released a CLI tool to transform MHTML to HTML files, see https://github.com/gildas-lormeau/mhtml-to-html
